MaxText/layers/attentions.py

```diff
@@ -632,6 +634,12 @@ def cudnn_flash_attention(
       mask_type = "padding_causal"  # only padding_causal mask type can take a created mask
       attn_mask = self.generate_attention_mask(query, key, decoder_segment_ids, model_mode)
 
+    # scaling factor for fused query weight or not
+    if self.query_pre_attn_scalar is None:
+      query_pre_attn_scalar = 1.0 / jnp.sqrt(query.shape[-1])
+    else:
+      query_pre_attn_scalar = self.query_pre_attn_scalar
```

Comment:
I don't think this is needed, as config.query_pre_attn_scalar is already used for scaling.